### ISSUE TITLE
"Illegal solution name" and no solution name entry area exists when creating a project.

### ISSUE THREAD

**Issue by aaronfranke** [mono/monodevelop#5453]
This isn't my main machine, and the same MonoDevelop version works fine on other machines. It's just this one machine which is misbehaving. Mono and MonoDevelop are installed through the official repos. Unfortunately there isn't too much information that I can provide, the error is strange.

Image of problem:

![k](https://user-images.githubusercontent.com/1646875/42914337-a4589e3c-8aae-11e8-8b71-403984b62e29.png)

System information and software versions:

![sdf](https://user-images.githubusercontent.com/1646875/42914340-a864044e-8aae-11e8-89f9-4f0b70962fb3.png)

    $ apt policy monodevelop
    monodevelop:
      Installed: 7.5.0.1254-0xamarin1+ubuntu1804b1

    $ mono --version
    Mono JIT compiler version 5.12.0.226 (tarball Thu May  3 09:42:09 UTC 2018)
    Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	    TLS:           __thread
	    SIGSEGV:       altstack
	    Notifications: epoll
	    Architecture:  amd64
	    Disabled:      none
	    Misc:          softdebug 
	    Interpreter:   yes
	    LLVM:          supported, not enabled.
	    GC:            sgen (concurrent by default)

Neofetch output:

    OS: Ubuntu 18.04 LTS x86_64 
    Host: Inspiron M5030 A03 
    Kernel: 4.15.0-23-generic 
    Uptime: 2 hours, 2 mins 
    Packages: 2818 
    Shell: bash 4.4.19 
    Resolution: 1366x768 
    DE: Xfce 
    WM: Xfwm4 
    WM Theme: Greybird-accessibility 
    Theme: Raleigh [GTK2], Greybird [GTK3] 
    Icons: Elementary-xfce-darker [GTK2/3] 
    Terminal: xfce4-terminal 
    Terminal Font: Monospace 12 
    CPU: AMD Athlon II P360 (2) @ 2.300GHz 
    GPU: AMD Mobility Radeon HD 4225/4250 
    Memory: 2475MiB / 3689MiB

**Comment by mrward** [comment]
So that error dialog appears when you click Next after selecting a project from the first page of the dialog? You do not see the final page of the New Project dialog?

Can you attach the IDE log after reproducing the problem? Help - Open Log Directory.

**Comment by aaronfranke** [comment]
@mrward Correct and correct. Here's the log as requested. [Ide.2018-07-19__01-17-24.log](https://github.com/mono/monodevelop/files/2208992/Ide.2018-07-19__01-17-24.log)

**Comment by mrward** [comment]
As it is trying to load the folder preview widget for the final page of the New Project dialog it looks like that is failing.

```
An unhandled exception has occured. Terminating MonoDevelop? False
System.MissingMethodException: bool Gtk.TreeIter.Equals(Gtk.TreeIter)
  at MonoDevelop.Ide.Projects.GtkProjectFolderPreviewWidget.AddProjectWithSolutionDirectoryToTree () [0x000c1] in <7136fc698a9f4040bfd15d8315767e5d>:0 
```

There is another missing method exception earlier on in the IDE log file.

```
An unhandled exception has occured. Terminating MonoDevelop? False
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.MissingMethodException: bool Gdk.Size.Equals(Gdk.Size)
  at (wrapper managed-to-native) 
```

So my guess is that there is something up with GTK# on that machine. If you were on the Mac I would suggest trying to uninstall Mono, remove the Mono directory to make sure everything is removed, and install Mono again to see if that fixes it. Not sure what the equivalent of that is on the Ubuntu - presumably there is some sort of `sudo apt remove` or can Mono be repaired?

**Comment by aaronfranke** [comment]
I tried `sudo apt purge mono-complete monodevelop` and re-installed, same error unfortunately.

**Comment by Therzok** [comment]
Which gtk# version do you have installed?

https://github.com/mono/gtk-sharp/commit/d2d422e22997461a948d8c4b131f32990c02c78f is where we added IEquatable to TreeIter.

**Comment by aaronfranke** [comment]
`sudo apt install gtk-sharp2` fixed the problem.

I think the issue is that the package `gtk-sharp2` is not listed as a dependency of MonoDevelop. If MonoDevelop requires `gtk-sharp2` then installing MonoDevelop should install `gtk-sharp2`.

**Comment by Therzok** [comment]
@directhex ^^

**Comment by aaronfranke** [comment]
Interestingly, uninstalling the `gtk-sharp2` package doesn't un-fix the issue. MonoDevelop continues to be fixed even after purging `gtk-sharp2*`.

`libgtk2.0-cil` was already installed by MonoDevelop while it was in the broken state.

I've attached the log but it's from installing MonoDevelop while it's in the fixed state. [dpkg.log](https://github.com/mono/monodevelop/files/2211442/dpkg.log)

So the fix is either adding `gtk-sharp2` as a dependency or making `libgtk2.0-cil` do what it's supposed to? Something that happened when installing `gtk-sharp2` fixed the issue for good on this machine.

More information, here's something I forgot to include. This machine was upgraded from Xubuntu 16.04 Xenial to Xubuntu 18.04 Bionic. The other machines, where it worked out of the box, are clean installs.

**Comment by directhex** [comment]
I guess it's only useful to see the log section from when it became fixed, i.e. I need older than the attached. dpkg.log.1 maybe?

**Comment by aaronfranke** [comment]
The old logs remove themselves when I shut down or reboot the computer. I did so several times to make sure there wasn't a file stuck in memory or something. I don't have the old logs, unfortunately.

I bet this issue could be reproduced if one upgraded Xenial to Bionic, then installed MonoDevelop. I may do this later, and get you all the logs. I would suggest not troubling yourself if you're busy.

**Comment by aaronfranke** [comment]
@rafabulsing Yep that's the same exact behavior I was experiencing. Please purge MonoDevelop, then install, then send your dpkg logs so that @directhex can review them. Also, what distro are you running, and did you upgrade it or clean install that version?

After you do that, you can also try installing `gtk-sharp3`. I didn't know which version MonoDevelop used at first so I tried that and it didn't work, then I installed `gtk-sharp2` after that. I'm not sure if that will work but it's the same procedure I used on my end.

**Comment by rafabulsing** [comment]
I'm on Ubuntu 18.04 LTS, and that was a clean install. Here's the log from the instalation (after purging, like you suggested):
[dpkg_monodevelopinstall.log](https://github.com/mono/monodevelop/files/2215778/dpkg_monodevelopinstall.log)

Installing `gtk-sharp3` did not help either, unfortunately. But still, thanks for the suggestion, it was worth a try!

**Comment by Therzok** [comment]
gtk-sharp2 is the one we use. Can you gist the logs of a session with this? I presume gtk# is not being picked up on your system

**Comment by rafabulsing** [comment]
Could you explain how I'd go about doing that? Sorry for my ignorance, I'm kind of a noob regarding that stuff. I'd love to help, though!

**Comment by Therzok** [comment]
[Logs can be found here](http://www.monodevelop.com/developers/articles/user-profiles/#logs)

If you see a stacktrace similar to the above, it means your gtk-sharp2 is too old and the monodevelop pkg got installed via pouring binaries, rather than source.

**Comment by rafabulsing** [comment]
Thanks for the pointer, sorry for the delay getting back to you. Here is the log:
[Ide.2018-07-22__13-38-17.log](https://github.com/mono/monodevelop/files/2217198/Ide.2018-07-22__13-38-17.log)

**Comment by Therzok** [comment]
Yup, it's the same issue. The gtk# 2.0 version you have installed is too old.

**Comment by rafabulsing** [comment]
I see. I'll try installing a more recent version of GTK#2, and see if that solves the issue. Thanks for the help!

**Comment by tiagossa1** [comment]
Thank you very much for the solution. I had the exact same problem. I tried to create any solution, I hit "next" and it happeard the error box saying "Illegal solution name". It was odd, since there's no solution name box to enter any name. I couldn't hit "Next", it was grayed out...

Pretty odd and for sure, it was a dependency problem, as wrote somewhere.

**Comment by DoctorChe** [comment]
I had the same problem.
It helped me to install **gtk-sharp2**.
Linux Mint 19 Tara (Cinnamon)
